### PR TITLE
[Task-6] Implement simple Express Online Shop application with Layered Architecture

### PR DESCRIPTION
**Pull request by Shagon1k**
**Postman collection to test:**
[Node.js course task-6.postman_collection.json](https://github.com/Shagon1k/nodejs-course/files/14695313/Node.js.course.task-6.postman_collection.json)

### CODE REVIEW

File: task-6/src/api/repositories/cart.repository.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  isDeleted: boolean;
+  items: ICartItemEntity[];
+}
+
+const carts_db: ICartEntity[] = [];
+
+export const findCartByUserId = (findUserId: string) => {
+  const cart = carts_db.find(({ userId }) => userId === findUserId) || null;
+
+  return structuredClone(cart);
```
Comment (Anton-Koniouchevsky) [inline task-6/src/api/repositories/cart.repository.ts:21]: praise: smart 

For real tasks please ensure that the node version used in all environments is suitable for such features. For example, the structured clone was added in v17 - https://nodejs.org/docs/latest/api/all.html#all_globals_structuredclonevalue-options

File: task-6/src/api/repositories/cart.repository.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  if (!cart) {
+    return;
+  }
+
+  const cartItemIndex = cart.items.findIndex(
+    ({ product: { id } }) => id === item.id
+  );
+
+  if (cartItemIndex === -1) {
+    cart.items.push({ product: item, count });
```
Comment (Anton-Koniouchevsky) [inline task-6/src/api/repositories/cart.repository.ts:67]: nitpick: if count === 0 this item is unnecessarily added to the cart

**Review by Anton-Koniouchevsky — APPROVED** [review]
Looks great in general!

File: task-6/src/api/repositories/cart.repository.ts (hunk 2 above)
Comment (Shagon1k) [reply task-6/src/api/repositories/cart.repository.ts:67]: Fixed :)

**Pull request merged**